python/test/test_energy_storage_device_wrappers.py

```diff
@@ -74,5 +74,40 @@ def test_sanity(self):
             device.evolve_one_time_step_constant_voltage(dt, U)
             self.assertAlmostEqual(device.get_voltage(), U)
 
+    def test_checkpoint_restart(self):
+        # check rc devices
+        for filename in valid_device_input[0:2]:
+            ptree = PropertyTree()
+            ptree.parse_info(filename)
+            device = EnergyStorageDevice(ptree)
+            dt = 0.1  # time_step in seconds
+            I = 5e-3  # current in amperes
+            device.evolve_one_time_step_constant_current(dt, I)
+            device.save('rc_device.txt')
+            new_device = EnergyStorageDevice(ptree)
+            new_device.load('rc_device.txt')
+            self.assertEqual(device.get_voltage(), new_device.get_voltage())
+            self.assertEqual(device.get_current(), new_device.get_current())
+        # check supercapacitor
+            coarse_mesh = 'coarse_mesh.z'
+            ptree = PropertyTree()
+            ptree.parse_info('super_capacitor.info')
+            ptree.put_bool('geometry.checkpoint', True)
+            ptree.put_string('geometry.coarse_mesh_filename', coarse_mesh)
+            device = EnergyStorageDevice(ptree)
+            dt = 0.1  # time_step in seconds
+            I = 5e-3  # current in amperes
+            device.evolve_one_time_step_constant_current(dt, I)
+            device.save('device.z')
+            ptree = PropertyTree()
+            ptree.parse_info('super_capacitor.info')
+            ptree.put_string('geometry.type', 'restart')
+            ptree.put_string('geometry.coarse_mesh_filename', coarse_mesh)
+            new_device = EnergyStorageDevice(ptree)
+            new_device.load('device.z')
+            self.assertEqual(device.get_voltage(), new_device.get_voltage())
+            self.assertEqual(device.get_current(), new_device.get_current())
```

Comment:
how does the error look like when you forget to set `geometry.checkpoint` to `True`?